src/System.Management.Automation/engine/parser/ast.cs

```diff
@@ -8406,7 +8408,8 @@ public override int GetHashCode()
         /// </remarks>
         internal bool IsType(Type type)
         {
-            string fullTypeName = type.FullName;
+            string? fullTypeName = type.FullName;
+            Debug.Assert(fullTypeName is not null);
```

Comment:
We probably should add a null check, but I left the behaviour unchanged for this PR.